src/node_jsvmapi.cc

```
@@ -1881,3 +1881,187 @@ napi_status napi_buffer_length(napi_env e, napi_value v, size_t* result) {
   *result = node::Buffer::Length(v8impl::V8LocalValueFromJsValue(v));
   return GET_RETURN_STATUS();
 }
+
+napi_status napi_is_arraybuffer(napi_env e, napi_value value, bool* result) {
+  NAPI_PREAMBLE(e);
+  CHECK_ARG(result);
+
+  v8::Local<v8::Value> v8value = v8impl::V8LocalValueFromJsValue(value);
+  *result = v8value->IsArrayBuffer();
+
+  return GET_RETURN_STATUS();
+}
+
+napi_status napi_create_arraybuffer(napi_env e,
+                                    size_t byte_length,
+                                    void** data,
+                                    napi_value* result) {
+  NAPI_PREAMBLE(e);
+  CHECK_ARG(result);
+
+  v8::Isolate *isolate = v8impl::V8IsolateFromJsEnv(e);
+  v8::Local<v8::ArrayBuffer> buffer;
+  if (data == nullptr || *data == nullptr) {
+    // Internal data, allocated and owned by the ArrayBuffer.
+    buffer = v8::ArrayBuffer::New(isolate, byte_length);
+
+    // The newly-allocated buffer is optionally returned.
+    if (data != nullptr) {
+      *data = buffer->GetContents().Data();
+    }
+  }
+  else {
+    // External data, not owned by the ArrayBuffer.
+    buffer = v8::ArrayBuffer::New(isolate, *data, byte_length);
+  }
+
+  *result = v8impl::JsValueFromV8LocalValue(buffer);
+  return GET_RETURN_STATUS();
+}
+
+napi_status napi_get_arraybuffer_info(napi_env e,
+                                      napi_value arraybuffer,
+                                      void** data,
+                                      size_t* byte_length) {
+  NAPI_PREAMBLE(e);
+
+  v8::Local<v8::Value> value = v8impl::V8LocalValueFromJsValue(arraybuffer);
+  RETURN_STATUS_IF_FALSE(value->IsArrayBuffer(), napi_invalid_arg);
+
+  v8::ArrayBuffer::Contents contents = value.As<v8::ArrayBuffer>()->GetContents();
+
+  if (data != nullptr) {
+    *data = contents.Data();
+  }
+
+  if (byte_length != nullptr) {
+    *byte_length = contents.ByteLength();
+  }
+
```

Should we return an error if both the data and byte_length values are nullptr ?